ports/glutin/window.rs

```diff
@@ -462,72 +452,57 @@ impl Window {
 
     fn handle_window_event(&self, event: glutin::Event) -> bool {
         match event {
-            Event::ReceivedCharacter(ch) => {
+            Event::WindowEvent { event: glutin::WindowEvent::ReceivedCharacter(ch), .. } => {
                 self.handle_received_character(ch)
             }
-            Event::KeyboardInput(element_state, _scan_code, Some(virtual_key_code)) => {
-                self.handle_keyboard_input(element_state, _scan_code, virtual_key_code);
+            Event::WindowEvent { event: glutin::WindowEvent::KeyboardInput(state, scan_code, Some(virtual_key_code), _), .. } => {
+                self.handle_keyboard_input(state, scan_code, virtual_key_code);
             }
-            Event::KeyboardInput(_, _, None) => {
+            Event::WindowEvent { event: glutin::WindowEvent::KeyboardInput(_, _, None, _), .. } => {
                 debug!("Keyboard input without virtual key.");
             }
-            Event::Resized(width, height) => {
+            Event::WindowEvent { event: glutin::WindowEvent::Resized(width, height), .. } => {
                 self.event_queue.borrow_mut().push(WindowEvent::Resize(TypedSize2D::new(width, height)));
             }
-            Event::MouseInput(element_state, mouse_button, pos) => {
+            Event::WindowEvent { event: glutin::WindowEvent::MouseInput(element_state, mouse_button), .. } => {
                 if mouse_button == MouseButton::Left ||
                    mouse_button == MouseButton::Right {
-                    match pos {
-                        Some((x, y)) => {
-                            self.mouse_pos.set(Point2D::new(x, y));
-                            self.event_queue.borrow_mut().push(
-                                WindowEvent::MouseWindowMoveEventClass(TypedPoint2D::new(x as f32, y as f32)));
-                            self.handle_mouse(mouse_button, element_state, x, y);
-                        }
-                        None => {
-                            let mouse_pos = self.mouse_pos.get();
-                            self.handle_mouse(mouse_button, element_state, mouse_pos.x, mouse_pos.y);
-                        }
-                    }
+                    let mouse_pos = self.mouse_pos.get();
+                    self.handle_mouse(mouse_button, element_state, mouse_pos.x, mouse_pos.y);
```

Comment:
We need `pos`. `mouse_pos` is updated on mouse move. But it's possible that the click happened without a prior mouse move event. For example, if the mouse changes position when servo was not focused.